File: from-2901-to-3000/2924-find-champion-ii/solution.go
func findChampion(n int, edges [][]int) int {
	incoming := make([]int, n)
	for _, edge := range edges {
		incoming[edge[1]] += 1
	}

	champion := -1
	for i := 0; i < n; i++ {
		if incoming[i] == 0 {
			if champion > -1 {
				return -1
			}
			champion = i
		}
	}
	return champion
}

File: from-2901-to-3000/2924-find-champion-ii/solution.py
class Solution:
    def findChampion(self, n: int, edges: List[List[int]]) -> int:
        incoming = [0] * n
        for _, dst in edges:
            incoming[dst] += 1

        champion = -1
        for i in range(n):
            if incoming[i] == 0:
                if champion > -1:
                    return -1
                else:
                    champion = i

        return champion